### PR TITLE
Ci-331 amend ro aao api to support ilr validation

### PR DESCRIPTION
**Pull request by mapaux**

### CODE REVIEW

File: src/Sfa.Das.ApprenticeshipInfoService.Api/Controllers/AssessmentOrgsController.cs [hunk 1]
```diff
@@ -133,9 +135,54 @@ public IEnumerable<Organisation> GetByStandardId(string id)
...
+                foreach (var standardOrganisation in response)
+                {
+                    standardOrganisation.Uri = ResolveStandardUri(standardOrganisation.StandardCode);
+                }
+
+                return response;
+            }
+            catch (Exception e)
+            {
+                _logger.Error(e, $"/assessment-organisations/{organisationId}/standars");
```
Comment (arvinthseran) [inline src/Sfa.Das.ApprenticeshipInfoService.Api/Controllers/AssessmentOrgsController.cs]: /standards?

File: src/Sfa.Das.ApprenticeshipInfoService.Api/Controllers/AssessmentOrgsController.cs [hunk 2]
```diff
@@ -133,9 +135,54 @@ public IEnumerable<Organisation> GetByStandardId(string id)
             return response;
         }
 
+        /// <summary>
+        /// Get standards by assessment organisation
+        /// </summary>
+        /// <returns>colllection of standards by specific organisation identifier</returns>
+        [SwaggerOperation("GetStandardsByOrganisationId")]
+        [SwaggerResponse(HttpStatusCode.OK, "OK", typeof(IEnumerable<StandardOrganisationSummary>))]
```
Comment (arvinthseran) [inline src/Sfa.Das.ApprenticeshipInfoService.Api/Controllers/AssessmentOrgsController.cs]: HttpStatusCode.NotFound?

File: src/Sfa.Das.ApprenticeshipInfoService.Infrastructure/Elasticsearch/AssessmentOrgsRepository.cs [hunk 3]
```diff
@@ -90,36 +93,32 @@ public IEnumerable<Organisation> GetOrganisationsByStandardId(string standardId)
                 throw new ApplicationException($"Failed query organisations by standard id");
             }
 
-            var organisations = results.Documents.Where(x => x.EffectiveFrom.Date <= DateTime.UtcNow.Date);
+            var organisations = results.Documents.Where(x => x.EffectiveFrom.Date <= DateTime.UtcNow.Date && (x.EffectiveTo >= DateTime.UtcNow.Date || x.EffectiveTo == null)).ToList();
```
Comment (arvinthseran) [inline src/Sfa.Das.ApprenticeshipInfoService.Infrastructure/Elasticsearch/AssessmentOrgsRepository.cs]: results.Documents.Where(x => ((x.EffectiveFrom != null) || x.EffectiveFrom.Date <= DateTime.UtcNow.Date) && (!x.EffectiveTo.HasValue || x.EffectiveTo.Value >= DateTime.UtcNow.Date)).ToList(); ?

looking at the excel file "Register - Standards" effective from can be null as well and we consider that scenario as not valid standard document.

File: src/Sfa.Das.ApprenticeshipInfoService.Api/Controllers/AssessmentOrgsController.cs (hunk 2 above)
Comment (scottcowan) [inline src/Sfa.Das.ApprenticeshipInfoService.Api/Controllers/AssessmentOrgsController.cs]: good point lets do that as another story as it's existing functionality

File: src/Sfa.Das.ApprenticeshipInfoService.Infrastructure/Elasticsearch/AssessmentOrgsRepository.cs (hunk 3 above)
Comment (mapaux) [inline src/Sfa.Das.ApprenticeshipInfoService.Infrastructure/Elasticsearch/AssessmentOrgsRepository.cs]: no need to check that, if EffectiveFrom date is null, it will not be indexed, because as you said we need a valid EffectiveFrom date

Comment (arvinthseran) [inline src/Sfa.Das.ApprenticeshipInfoService.Infrastructure/Elasticsearch/AssessmentOrgsRepository.cs]: i will add an excel tests to check for an empty effective from date. so we don't have to carry the validation in our api.

File: src/Sfa.Das.ApprenticeshipInfoService.Api/Controllers/AssessmentOrgsController.cs [hunk 4]
```diff
@@ -133,9 +135,54 @@ public IEnumerable<Organisation> GetByStandardId(string id)
...
+        [Route("assessment-organisations/{organisationId}/standards", Name = "GetStandardsByOrganisationId")]
+        [ExceptionHandling]
+        public IEnumerable<StandardOrganisationSummary> GetStandardsByOrganisationId(string organisationId)
+        {
+            try
+            {
+                var response = _getAssessmentOrgs.GetStandardsByOrganisationIdentifier(organisationId).ToList();
```
Comment (arvinthseran) [inline src/Sfa.Das.ApprenticeshipInfoService.Api/Controllers/AssessmentOrgsController.cs]: `if (_getStandards.GetOrganisationById(id) == null)
{
throw HttpResponseFactory.RaiseException(
HttpStatusCode.NotFound,
$"No organisation with EpaOrganisationIdentifier {id} found");
}`

can we add this logic?.

File: src/Sfa.Das.ApprenticeshipInfoService.Api/Controllers/AssessmentOrgsController.cs [hunk 5]
```diff
@@ -133,9 +135,54 @@ public IEnumerable<Organisation> GetByStandardId(string id)
...
+        /// Get standards by assessment organisation
+        /// </summary>
+        /// <returns>colllection of standards by specific organisation identifier</returns>
+        [SwaggerOperation("GetStandardsByOrganisationId")]
+        [SwaggerResponse(HttpStatusCode.OK, "OK", typeof(IEnumerable<StandardOrganisationSummary>))]
+        [Route("assessment-organisations/{organisationId}/standards", Name = "GetStandardsByOrganisationId")]
+        [ExceptionHandling]
+        public IEnumerable<StandardOrganisationSummary> GetStandardsByOrganisationId(string organisationId)
+        {
+            try
```
Comment (scottcowan) [inline src/Sfa.Das.ApprenticeshipInfoService.Api/Controllers/AssessmentOrgsController.cs]: can you remove the try/catch here, it'll get logged higher up

File: src/Sfa.Das.ApprenticeshipInfoService.Infrastructure/Elasticsearch/AssessmentOrgsRepository.cs (hunk 3 above)
Comment (scottcowan) [inline src/Sfa.Das.ApprenticeshipInfoService.Infrastructure/Elasticsearch/AssessmentOrgsRepository.cs]: It should also be 
```c#
x.EffectiveTo.Date
```

Comment (scottcowan) [inline src/Sfa.Das.ApprenticeshipInfoService.Infrastructure/Elasticsearch/AssessmentOrgsRepository.cs]: It's ok if there's empty EffectiveFrom date, since there's some in there already. we can talk to richard about removing those before we add a test to prevent it
